pkgs/applications/networking/instant-messengers/slack/default.nix

```diff
@@ -161,6 +162,18 @@ let
       defaults write com.tinyspeck.slackmacgap SlackNoAutoUpdates -bool YES
     '';
   };
-in if stdenv.isDarwin
-  then darwin
-  else linux
+
+  # Fallback to dummy derivation to give the user a better error message
+  dummyDerivation = runCommand pname {
+    name = pname;
+    inherit pname version meta;
+    platforms = [];
+  } ''
+    >&2 echo "There is no Slack derivation for system ${system}"
+    exit 1
+  '';
+
+in {
+  x86_64-darwin = darwin;
+  x86_64-linux = linux;
+}.${system} or dummyDerivation
```

Comment:
```suggestion
}.${system} or throw "No slack derivation for ${stdenv.hostPlatform.system}"
```

Reply:
This is still not friendly enough IMO and inconsistent with how other derivations behave when trying to build for an unsupported platform.

Simply throwing an error will cause the package not to show up in `nix-env -qaP` which is confusing.

Reply:
I wouldn't say its inconsistent with how other derivations handle building for unsupported platforms. In addition to setting `meta.plaforms`, `throw`ing a "unsupported platform" message seems quite common:

 - pkgs/applications/graphics/unigine-valley/default.nix
 - pkgs/applications/misc/playonlinux/default.nix
 - pkgs/development/compilers/cmucl/binary.nix
 - pkgs/development/compilers/openjdk/openjfx/11.nix
 - pkgs/development/compilers/openjdk/openjfx/13.nix
 - pkgs/development/compilers/openjdk/8.nix
 - pkgs/development/compilers/sagittarius-scheme/default.nix
 - pkgs/development/libraries/xalanc/default.nix
 - pkgs/development/pharo/vm/build-vm.nix
 - pkgs/development/tools/poetry2nix/poetry2nix/pep508.nix
 - pkgs/games/steam/default.nix
 - pkgs/games/vessel/default.nix
 - pkgs/shells/powershell/default.nix
 - pkgs/tools/security/enpass/default.nix
 - pkgs/top-level/lua-packages.nix

There's even a  `x = {…}.${system} or throwSystem;` pattern used in some places:

 - pkgs/applications/blockchains/mist.nix
 - pkgs/applications/misc/simplenote/default.nix
 - pkgs/applications/networking/instant-messengers/slack/default.nix
 - pkgs/applications/networking/instant-messengers/wire-desktop/default.nix
 - pkgs/development/libraries/oracle-instantclient/default.nix
 - pkgs/development/tools/electron/3.x.nix

Building a dummy derivation, only for the purpose of it failing a build looks more hacky to me.